### ISSUE TITLE
bison: error: '%name-prefix' and '%define api.prefix' cannot be used together

### ISSUE THREAD

**Issue by dl8dtl** [doxygen/doxygen#10898]
**Describe the bug**
Compiling 1.11.0 on FreeBSD, I eventually get
```
[ 27%] Generating ../generated_src/ce_parse.h
cd /usr/ports/devel/doxygen/work/.build/generated_src && /usr/local/bin/bison -l -d -p ce_parsexpYY /usr/ports/devel/doxygen/work/doxygen-1.11.0/src/constexp.y -o ce_parse.c
/usr/ports/devel/doxygen/work/doxygen-1.11.0/src/constexp.y:36.1-31: error: '%name-prefix' and '%define api.prefix' cannot be used together
   36 | %define api.prefix {constexpYY}
      | ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
*** Error code 1
```

**To Reproduce**
Compile on FreeBSD

**Expected behavior**
It compiles :-)

**Version**
1.11.0 source

**Additional context**
I think the `-p` option is not needed when using `%define api.prefix`, and could go away:
```
--- src/CMakeLists.txt.orig     2024-05-10 21:34:45.000000000 +0200
+++ src/CMakeLists.txt  2024-05-26 16:45:05.376959000 +0200
@@ -72,7 +71,7 @@
 
 # ce_parse.h
 add_custom_command(
-    COMMAND ${BISON_EXECUTABLE} -l -d -p ce_parsexpYY ${CMAKE_CURRENT_LIST_DIR}/constexp.y -o ce_parse.c
+    COMMAND ${BISON_EXECUTABLE} -l -d ${CMAKE_CURRENT_LIST_DIR}/constexp.y -o ce_parse.c
     DEPENDS ${CMAKE_CURRENT_LIST_DIR}/constexp.y
     OUTPUT ${GENERATED_SRC}/ce_parse.h
     WORKING_DIRECTORY ${GENERATED_SRC}
```

**Comment by albert-github** [comment]
- Which version of bison are you using?
- Which version of flex are you using?
- do you have special settings so that an error is thrown and not a warning is given as on MacOS and Ubuntu (GitHub Actions) I see something like:
  ```
  [ 31%] Generating ../generated_src/ce_parse.h
  /Users/runner/work/doxygen/doxygen/src/constexp.y:36.1-25: warning: deprecated directive: ‘%name-prefix "constexpYY"’, use ‘%define api.prefix {constexpYY}’ [-Wdeprecated]
     36 | %name-prefix "constexpYY"
        | ^~~~~~~~~~~~~~~~~~~~~~~~~
        | %define api.prefix {constexpYY}
  ```
I tried it on Windows with the MSVC compiler (using bison 2.7 and flex version 2.6.4) and it looks like to be OK.
I checked the versions on GitHub actions it looks like Ubuntu and MacOS use the same or even newer versions, but on Windows the flex version 2.6.3 is used.
The CMakeLists.txt specifies that flex should be `>= 2.5.37` so maybe this could be changed to `>= 2.6.3

**Comment by dl8dtl** [comment]
> Which version of bison are you using?

3.8.2

> * Which version of flex are you using?

2.6.4 (Bundled in FreeBSD's base system)



> * do you have special settings

Not that I know of.

Anyway, bison's documentation says that -p is deprecated in favor of the `%define`, so why not drop it?

**Comment by albert-github** [comment]
> Anyway, bison's documentation says that -p is deprecated in favor of the %define, so why not drop it?

This is valid for the newer versions and seen the fact that the minimum required version of bison is 2.7 I think it would be possible to drop the constraint.

I've just pushed a proposed patch, pull request #10899
(also removing an old test on the flex version and updating the minimum flex version)

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.12.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
